src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/AssociatedFilingTransactionMapper.java

```diff
@@ -51,9 +62,19 @@ public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(Inter
                                             associatedFilingChildMapper.mapChild(associatedFiling, request);
                                         },
                                         // Add new associated filing to existing list
-                                        () -> associatedFilingList
-                                                .add(associatedFilingChildMapper
-                                                        .mapChild(new FilingHistoryAssociatedFiling(), request))),
+                                        () -> {
+                                            if (associatedFilingList.stream()
+                                                    .anyMatch(associatedFiling ->
+                                                            StringUtils.isBlank(associatedFiling.getEntityId()))) {
+                                                LOGGER.error(
```

Comment:
Is this really an error? It will be a normal event to annotate a filing history document created by the Perl code, so can this log message be changed to `info`? Errors should really be reserved for something that Fess should react to when we go Live.